packages/aws-rfdk/lib/deadline/lib/thinkbox-docker-images.ts

```diff
@@ -0,0 +1,280 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { randomBytes } from 'crypto';
+import * as path from 'path';
+
+import {
+  ContainerImage,
+  RepositoryImage,
+} from '@aws-cdk/aws-ecs';
+import {
+  Code,
+  SingletonFunction,
+  Runtime,
+} from '@aws-cdk/aws-lambda';
+import { RetentionDays } from '@aws-cdk/aws-logs';
+import {
+  Construct,
+  CustomResource,
+  Duration,
+  Token,
+} from '@aws-cdk/core';
+
+import {
+  IVersion,
+  RenderQueueImages,
+  ThinkboxManagedDeadlineDockerRecipes,
+  UsageBasedLicensingImages,
+} from '.';
+
+/**
+ * Choices for signifying the user's stance on the terms of the AWS Thinkbox End-User License Agreement (EULA).
+ * See: https://www.awsthinkbox.com/end-user-license-agreement
+ */
+export enum AwsThinkboxEulaAcceptance {
+  /**
+   * The user signifies their explicit rejection of the tems of the AWS Thinkbox EULA.
+   *
+   * See: https://www.awsthinkbox.com/end-user-license-agreement
+   */
+  USER_REJECTS_AWS_THINKBOX_EULA = 0,
+
+  /**
+   * The user signifies their explicit acceptance of the terms of the AWS Thinkbox EULA.
+   *
+   * See: https://www.awsthinkbox.com/end-user-license-agreement
+   */
+  USER_ACCEPTS_AWS_THINKBOX_EULA = 1,
+}
+
+/**
+ * Interface to specify the properties when instantiating a {@link ThinkboxDockerImages} instnace.
+ */
+export interface ThinkboxDockerImagesProps {
+  /**
+   * The Deadline version to obtain images for.
+   * @default latest
+   */
+  readonly version?: IVersion;
+
+  /**
+   * Deadline is licensed under the terms of the AWS Thinkbox End-User License Agreement (see: https://www.awsthinkbox.com/end-user-license-agreement).
+   * Users of ThinkboxDockerImages must explicitly signify their acceptance of the terms of the AWS Thinkbox EULA through this
+   * property before the {@link ThinkboxDockerImages} will be allowed to deploy Deadline.
+   */
+  // Developer note: It is a legal requirement that the default be USER_REJECTS_AWS_THINKBOX_EULA.
+  readonly userAwsThinkboxEulaAcceptance: AwsThinkboxEulaAcceptance;
+}
+
+/**
+ * An API for interacting with publicly available Deadline container images published by AWS Thinkbox.
```

Comment:
Not sure if it matters, but in most cases you've used 
```suggestion
 * An API for interacting with publicly available AWS Thinkbox Deadline container images.
```

Reply:
I think this can be left as-is unless you strongly object.